dev/PipsControl/PipsControl.cpp

```diff
@@ -0,0 +1,335 @@
+
+    [this](const winrt::Grid grid) {
+        if (grid) {
+            m_rootGridPointerEnteredRevoker = grid.PointerEntered(winrt::auto_revoke, { this, &PipsControl::OnPipsControlPointerEntered });
+            m_rootGridPointerExitedRevoker = grid.PointerExited(winrt::auto_revoke, { this, &PipsControl::OnPipsControlPointerExited });
+            m_rootGridKeyDownRevoker = grid.KeyDown(winrt::auto_revoke, { this, &PipsControl::OnRootGridKeyDown });
+
+        }
+    }(GetTemplateChildT<winrt::Grid>(c_rootGridName, controlProtected));
+
+    OnNumberOfPagesChanged(0);
+    OnSelectedPageIndexChange(-1);
+    OnMaxDisplayedPagesChanged(0);
+}
+
+void PipsControl::OnPipsControlPointerEntered(winrt::IInspectable sender, winrt::PointerRoutedEventArgs args) {
+    m_isPointerOver = true;
+    UpdateNavigationButtonVisualStates();
+}
+void PipsControl::OnPipsControlPointerExited(winrt::IInspectable sender, winrt::PointerRoutedEventArgs args) {
+    // TODO : Explain WHAT'S UP and mention quarks
+    if (!isWithinBounds(args.GetCurrentPoint(*this).Position())) {
+        m_isPointerOver = false;
+        HideNavigationButtons();
+        args.Handled(true);
+    }
+
+}
+
+bool PipsControl::isWithinBounds(winrt::Point point) {
+    return point.X >= 0 && point.X <= ActualWidth() && point.Y >= 0 && point.Y <= ActualHeight();
+}
+
+void PipsControl::HideNavigationButtons() {
+    winrt::VisualStateManager::GoToState(*this, c_previousPageButtonHiddenVisualState, false);
+    winrt::VisualStateManager::GoToState(*this, c_nextPageButtonHiddenVisualState, false);
+}
+void PipsControl::UpdateNavigationButtonVisualStates() {
+
+    const int selectedPageIndex = SelectedPageIndex();
+    const int numberOfPages = NumberOfPages();
+    const int maxDisplayedPages = MaxDisplayedPages();
+
+    if (!numberOfPages == 0 && maxDisplayedPages > 0) {
+        if (selectedPageIndex != 0) {
+            if (m_isPointerOver) {
+                winrt::VisualStateManager::GoToState(*this, c_previousPageButtonVisibleVisualState, false);
+            }
+            winrt::VisualStateManager::GoToState(*this, c_previousPageButtonEnabledVisualState, false);
+        }
+        else {
+            winrt::VisualStateManager::GoToState(*this, c_previousPageButtonHiddenVisualState, false);
+            winrt::VisualStateManager::GoToState(*this, c_previousPageButtonDisabledVisualState, false);
+        }
+
+        if (selectedPageIndex != numberOfPages - 1) {
+            if (m_isPointerOver) {
+                winrt::VisualStateManager::GoToState(*this, c_nextPageButtonVisibleVisualState, false);
+            }
+            winrt::VisualStateManager::GoToState(*this, c_nextPageButtonEnabledVisualState, false);
+        }
+        else {
+            winrt::VisualStateManager::GoToState(*this, c_nextPageButtonHiddenVisualState, false);
+            winrt::VisualStateManager::GoToState(*this, c_nextPageButtonDisabledVisualState, false);
+        }
+    }
```

Comment:
This block is a bit hard to read. Can you write a comment or come up with a simpler logic to pick the states ?

Reply:
Modified the logic a little bit:) let me know if that's better